### PR TITLE
#162414168 create article comment

### PR DESCRIPTION
**Pull request by Luckzman**
#### What does this PR do?
This PR creates and get articles comment.

#### Description of Task to be completed?
Based on the fact that users need to comment on articles, there is need for user to create comment and read comments.
#### How should this be manually tested?
`run npm test`
What are the relevant pivotal tracker stories?
PT Link: https://www.pivotaltracker.com/story/show/162414165

Screenshots
screenshot 2019-01-25 at 10 41 01 am

### CODE REVIEW

File: test/modelmanagers/articlemodel.test.js [hunk 1]
```diff
@@ -42,8 +47,9 @@ describe('Unit test article model manager functions', () => {
     expect(updatedDetails[0].dataValues.title).to.be.equals(updatedArticle.title);
     expect(updatedDetails[0].dataValues.content).to.be.equals(updatedArticle.content);
     expect(updatedDetails[0].dataValues.description).to.be.equals(updatedArticle.description);
-    expect(updatedDetails[0].dataValues.featuredImageUrl)
-      .to.be.equals(updatedArticle.featuredImageUrl);
+    expect(updatedDetails[0].dataValues.featuredImageUrl).to.be.equals(
+      updatedArticle.featuredImageUrl
+    );
```
[the remark this answers is not in the document]
Comment (iMichaelOwolabi) [reply test/modelmanagers/articlemodel.test.js]: Kindly fix hound issues

**Review by iMichaelOwolabi — DISMISSED** [review]

**Review by iAfolayan — DISMISSED** [review]

**Review by iAfolayan — DISMISSED** [review]

File: controllers/articlesCommentController.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * @param {*} req
+ * @param {*} res
+ * @returns {*} *
+ */
+async function findCommentController(req, res) {
+  try {
+    const { articleId } = req.params;
+    const { q } = req.query;
+    const comments = await Comment.findAllComment(articleId, q);
+    comments.map((thread) => {
```
Comment (codrex) [inline controllers/articlesCommentController.js]: Array.map is not a mutating function. You are calling dot map on comments and not assigning the array returned to any variable. The result of this is that your comments will remain unchanged and the array return from the map operation will be garbage collected and the whole operation would have been for nothing. Please assign the returned array to a variable and send that to the user as comments.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

File: controllers/articlesCommentController.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ *
+ *
+ * @param {*} req
+ * @param {*} res
+ * @returns {*} *
+ */
+async function findCommentThreadController(req, res) {
+  try {
+    const { articleId, commentId } = req.params;
+    const { q } = req.query;
```
Comment (codrex) [inline controllers/articlesCommentController.js]: what is q?

File: lib/modelManagers/articlesComment.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
+// const user = require('../../database/models').User;
+// const articles = require('../../database/models');
```
Comment (codrex) [inline lib/modelManagers/articlesComment.js]: remove dead code

File: lib/modelManagers/articlesComment.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        mainCommentId: {
+          [Op.eq]: null
+        },
+        articleId
+      },
+      include: {
+        model: articlesComment,
+        as: 'threads'
+      },
+      limit: 5 * Number(offset)
```
Comment (codrex) [inline lib/modelManagers/articlesComment.js]: Please make you of the pagination function available

File: lib/modelManagers/articlesComment.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+   * @memberof articleComment
+   */
+  static async findCommentThreads(articleId, commentId, offset = 1) {
+    const options = {
+      where: {
+        id: {
+          [Op.eq]: commentId
+        },
+        articleId
+      },
```
Comment (codrex) [inline lib/modelManagers/articlesComment.js:80]: Can we have only main comment id returned here?

File: lib/modelManagers/articlesComment.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      include: {
+        model: articlesComment,
+        as: 'threads',
+        limit: 5 * Number(offset)
```
Comment (codrex) [inline lib/modelManagers/articlesComment.js]: same ☝️

File: lib/utils/commentValidator.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
+const { validateComment } = require('./validate'),
+  { failureResponse } = require('../utils/messageHandler'),
+  { BAD_REQUEST_CODE } = require('../../constants/index'),
+  formatErrorMessage = require('../utils/messageFormatter');
+
+const commentValidator = (req, res, next) => {
+  validateComment(req, 'comment');
```
Comment (codrex) [inline lib/utils/commentValidator.js]: `validateComment` does not accept a second parameter

File: routes/api/articlesRoute.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
+const router = require('express').Router(),
+  //   jwtValidator = require('../../middlewares/jwtValidator'),
```
Comment (codrex) [inline routes/api/articlesRoute.js]: dead code

File: controllers/articlesCommentController.js (hunk 2 above)
Comment (Luckzman) [reply controllers/articlesCommentController.js]: fixed

File: controllers/articlesCommentController.js (hunk 3 above)
Comment (Luckzman) [reply controllers/articlesCommentController.js]: it represents the query variable that will be passed in from the postman or web browser which represent the offset for pagination.

File: lib/modelManagers/articlesComment.js (hunk 6 above)
Comment (Luckzman) [reply lib/modelManagers/articlesComment.js:80]: yes the commentId is the mainCommentId in the params that must match with the Id of the comment to be searched for

File: controllers/articlesCommentController.js (hunk 3 above)
Comment (codrex) [reply controllers/articlesCommentController.js]: Please use a descriptive variable name

File: lib/modelManagers/articlesComment.js (hunk 7 above)
Comment (Luckzman) [reply lib/modelManagers/articlesComment.js]: fixed

File: lib/modelManagers/articlesComment.js (hunk 5 above)
Comment (Luckzman) [reply lib/modelManagers/articlesComment.js]: fixed

File: controllers/articlesCommentController.js (hunk 3 above)
Comment (Luckzman) [reply controllers/articlesCommentController.js]: done

**Review by tosinibrahim96 — DISMISSED** [review]

**Review by iAfolayan — DISMISSED** [review]

**Review by Maxfurry — DISMISSED** [review]

File: controllers/articlesCommentController.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+ * @returns {*} *
+ */
+async function findCommentThreadController(req, res) {
+  try {
+    const { articleId, mainCommentId } = req.params;
+    let { page } = req.query;
+    if (!page) {
+      page = 1;
+    }
+    const query = { offset: page / page, limit: COMMENT_LIMIT * page };
```
Comment (codrex) [inline controllers/articlesCommentController.js]: ` const query = { offset: page / page, limit: COMMENT_LIMIT * page };` Please I would love to know what you are trying to achieve here

**Review by codrex — APPROVED** [review]

**Review by ibraheemkabir — APPROVED** [review]

**Pull request merged**